components/brave_wallet_ui/async/wallet_panel_async_handler.ts

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+import { MiddlewareAPI, Dispatch, AnyAction } from 'redux'
+import AsyncActionHandler from '../../common/AsyncActionHandler'
+import * as Actions from '../actions/wallet_panel_actions'
+import { APIProxy, WalletPanelApiProxy, State, WalletPanelReducerState } from '../constants/types'
+import { AccountPayloadType } from '../constants/action_types'
+
+const walletPanelApiProxyImpl: APIProxy = require('../wallet_panel_api_proxy.js').default
+const handler = new AsyncActionHandler()
+
+function getState (store: MiddlewareAPI<Dispatch<AnyAction>, any>): WalletPanelReducerState {
+  return (store.getState() as State).walletPanelReducer
+}
+
+handler.on(Actions.initialize.getType(), async (store) => {
+  const state = getState(store)
+  // Don't initialize more than once.
+  if (!state.hasInitialized) {
+    store.dispatch(Actions.initialized({
+      isConnected: false
+    }))
+  }
+})
+
+handler.on(Actions.cancelConnectToSite.getType(), async (store) => {
+  const apiProxy: WalletPanelApiProxy = walletPanelApiProxyImpl.getInstance()
+  console.log('cancel connect to site!')
+  apiProxy.closeUI()
+})
+
+handler.on(Actions.connectToSite.getType(), async (store, payload: AccountPayloadType) => {
+  const apiProxy: WalletPanelApiProxy = walletPanelApiProxyImpl.getInstance()
+  console.log('Got connect to site for accounts: ', payload.selectedAccounts, 'for site:', payload.siteToConnectTo)
+  apiProxy.closeUI()
+})
+
+handler.on(Actions.visibilityChanged.getType(), (store) => {
+  const apiProxy: WalletPanelApiProxy = walletPanelApiProxyImpl.getInstance()
+  apiProxy.showUI()
```

Comment:
Is the purpose of calling `apiProxy.showUI` on initialization to have the bubble only actually show when the content is first ready?

Reply:
the popup just won't appear without it because no visibilityChanged would be fired until it's open.  But it does get fired on the second click. Not sure why but the tab search polymer stuff does it the same way so I just got it from there. I tried to simplify it with no luck.